core/frontend.ml

```diff
@@ -54,6 +53,7 @@ struct
       let _program = CheckXmlQuasiquotes.checker#program program in
       let () = DesugarSessionExceptions.settings_check program in
       ((( ExperimentalExtensions.check#program
+       ->- DesugarSessionExceptions.sanitise_linear_handlers
```

Comment:
This should **technically** only be run whenever `session_exceptions` is enabled.